client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -606,14 +627,45 @@ export default {
           const mergedPage = new compose.Page({ namespaceID, ...page, blocks: this.blocks })
 
           this.updatePage(mergedPage).then((page) => {
-            this.unsavedBlocks.clear()
-            this.toastSuccess(this.$t('notification:page.saved'))
-            if (closeOnSuccess) {
-              this.$router.push({ name: 'admin.pages' })
-            } else if (previewOnSuccess) {
-              this.$router.push({ name: 'page', params: { pageID: this.pageID } })
+            const hasTempID = this.page.blocks.filter(b => b.kind === 'Tabs').some(b => b.options.tabs.some(t => t.blockID.length > 8))
```

Comment:
t.blockID.length > 8 is not a good condition to check that, consider prefixing the tempID with `tempID-`. So that you can know for sure

Also instead of doing the computation twice, why not just do the tabsWithTempIDs and check if length > 0?
Otherwise, you're looping twice through the arrays